airbyte-integrations/connectors/source-google-sheets/google_sheets_source/helpers.py

```diff
@@ -30,7 +31,10 @@ def get_authenticated_drive_client(credentials: Dict[str, str], scopes: List[str
 
     @staticmethod
     def get_authenticated_google_credentials(credentials: Dict[str, str], scopes: List[str] = SCOPES):
-        return service_account.Credentials.from_service_account_info(credentials, scopes=scopes)
+        if credentials.get("auth_type") == "Service":
+            return service_account.Credentials.from_service_account_info(json.loads(credentials["service_account_info"]), scopes=scopes)
+        elif credentials.pop("auth_type") == "Client":
+            return client_account.Credentials.from_authorized_user_info(info=credentials)
```

Comment:
I might be wrong, but what will happen here if you first comparison will be False?
so the `auth_type` is already removed, would it always return None?

Reply:
ok, I read it as a two `pop`s, but nevertheless can't we pop `auth_type` first and then do if/else logic?